rts/System/Ecs/Components/BaseComponents.h

```diff
@@ -18,14 +20,23 @@ struct BasicComponentType<entt::entity> {
 #define ALIAS_COMPONENT_DEF(Component, T, DefaultValue) \
 struct Component : public BasicComponentType<T> { \
     Component(){ value = DefaultValue; } \
-    Component(T val){ value = val; } \
+    Component(T val){ value = std::move(val); } \
```

Comment:
perhaps this could go even further and initialize `value` directly via an initializer list
```cpp
Component() : value(DefaultValue) { } \
Component(T val) : value(std::move(val)) { } \
```

Reply:
You are quite right.

Reply:
Ah actually, I can't because Component isn't the owner of value. BasicComponentType<T> is. So Component can't initialize value in its initializer list.

Reply:
then how about
```
Component(T val) : BasicComponent(std::move(val)) { }
```

Reply:
Not as clean as I would like because I have to add constructors to the BasicComponentType defintions (there may be more of those in future if different default/initial values have to be specified), but sure I can do that.

Reply:
nah if the compiler does not provide a default constructor then don't worry about it

Reply:
The compiler provides a default constructor, but not a constructor that takes an argument. If you specify that then you have to also provide the default constructor yourself as well.